runtimes/common/src/dip/merkle.rs

```diff
@@ -274,30 +293,27 @@ pub mod v0 {
 					.ok_or(DidMerkleProofError::KeyNotFound)?;
 				// Create the merkle leaf key depending on the relationship of the key to the
 				// DID document.
-				let did_key_merkle_key: DidKeyMerkleKey<KeyIdOf<Runtime>> = if *key_id == did_details.authentication_key
-				{
-					Ok((*key_id, DidVerificationKeyRelationship::Authentication.into()).into())
+				let key_relationship = if *key_id == did_details.authentication_key {
+					Ok(DidVerificationKeyRelationship::Authentication.into())
 				} else if Some(*key_id) == did_details.attestation_key {
-					Ok((*key_id, DidVerificationKeyRelationship::AssertionMethod.into()).into())
+					Ok(DidVerificationKeyRelationship::AssertionMethod.into())
 				} else if Some(*key_id) == did_details.delegation_key {
-					Ok((*key_id, DidVerificationKeyRelationship::CapabilityDelegation.into()).into())
+					Ok(DidVerificationKeyRelationship::CapabilityDelegation.into())
 				} else if did_details.key_agreement_keys.contains(key_id) {
-					Ok((*key_id, DidKeyRelationship::Encryption).into())
+					Ok(DidKeyRelationship::Encryption)
 				} else {
 					log::error!("Unknown key ID {:#?} retrieved from DID details.", key_id);
 					Err(DidMerkleProofError::Internal)
 				}?;
```

Comment:
is a match statement not better suited?

Reply:
The conditions being tested are of different nature. E.g., `else if Some(*key_id) == did_details.attestation_key` and `else if did_details.key_agreement_keys.contains(key_id)`. As far as I know you can't put them in the same match statement without trying to match on both conditions at one. I don't think it would make it more readable...

Reply:
Ah yeah, I didn't see the `contains` in key_agreement_keys.